{"isMarkdown":true,"isTranslated":false,"isActive":true,"flaws":{},"title":"Generator.prototype.throw()","mdn_url":"/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator/throw","locale":"en-US","native":"English (US)","browserCompat":["javascript.builtins.Generator.throw"],"summary":"The throw() method of Generator instances acts as if a throw statement is inserted in the generator's body at the current suspended position, which informs the generator of an error condition and allows it to handle the error, or perform cleanup and close itself.","popularity":0.0012,"modified":"2023-09-25T07:33:31.000Z","source":{"folder":"en-us/web/javascript/reference/global_objects/generator/throw","github_url":"https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/generator/throw/index.md","last_commit_url":"https://github.com/mdn/content/commit/fb85334ffa4a2c88d209b1074909bee0e0abd57a","filename":"index.md"},"short_title":"Generator.prototype.throw()","parents":[{"uri":"/en-US/docs/Web","title":"References"},{"uri":"/en-US/docs/Web/JavaScript","title":"JavaScript"},{"uri":"/en-US/docs/Web/JavaScript/Reference","title":"Reference"},{"uri":"/en-US/docs/Web/JavaScript/Reference/Global_Objects","title":"Standard built-in objects"},{"uri":"/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator","title":"Generator"},{"uri":"/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator/throw","title":"Generator.prototype.throw()"}],"pageTitle":"Generator.prototype.throw() - JavaScript | MDN","noIndexing":false,"hash":"651c1f62ed7413431411e675c6c469d0832f442877795592605fe39e140702ce"}